lib/oli/resources/collaboration.ex

```diff
@@ -339,6 +340,97 @@ defmodule Oli.Resources.Collaboration do
     )
   end
 
+  @doc """
+  Returns the list of posts created by an user in a section.
+
+  ## Examples
+
+      iex> list_last_posts_for_user(1, 1, 5)
+      [%Post{status: :archived}, ...]
+
+      iex> list_last_posts_for_user(2, 2, 10)
+      []
+  """
+
+  def list_lasts_posts_for_user(user_id, section_id, limit) do
+    Repo.all(
+      from(
+        post in Post,
+        join: sr in SectionResource,
+        on: sr.resource_id == post.resource_id and sr.section_id == post.section_id,
+        join: spp in SectionsProjectsPublications,
+        on: spp.section_id == post.section_id and spp.project_id == sr.project_id,
+        join: pr in PublishedResource,
+        on: pr.publication_id == spp.publication_id and pr.resource_id == post.resource_id,
+        join: rev in Revision,
+        on: rev.id == pr.revision_id,
+        left_join: rev2 in Revision,
+        on: rev.resource_id == rev2.resource_id and rev.id < rev2.id,
+        join: user in User,
+        on: post.user_id == user.id,
+        where:
+          post.section_id == ^section_id and post.user_id == ^user_id and
+            (post.status in [:approved, :archived] or
+               (post.status == :submitted and post.user_id == ^user_id)) and is_nil(rev2) and
+            rev.deleted == false,
+        select: %{
+          id: post.id,
+          content: post.content,
+          user_name: user.name,
+          title: rev.title,
+          updated_at: post.updated_at
+        },
+        order_by: [desc: :inserted_at],
+        limit: ^limit
+      )
+    )
+  end
+
+  @doc """
+  Returns the list of posts created in all resources of section.
+
+  ## Examples
+
+      iex> list_last_posts_for_section(1, 1, 5)
+      [%Post{status: :archived}, ...]
+
+      iex> list_last_posts_for_section(2, 2, 10)
+      []
+  """
+
+  def list_lasts_posts_for_section(user_id, section_id, limit) do
+    Repo.all(
+      from(
+        post in Post,
+        join: sr in SectionResource,
+        on: sr.resource_id == post.resource_id and sr.section_id == post.section_id,
+        join: spp in SectionsProjectsPublications,
+        on: spp.section_id == post.section_id and spp.project_id == sr.project_id,
+        join: pr in PublishedResource,
+        on: pr.publication_id == spp.publication_id and pr.resource_id == post.resource_id,
+        join: rev in Revision,
+        on: rev.id == pr.revision_id,
+        left_join: rev2 in Revision,
+        on: rev.resource_id == rev2.resource_id and rev.id < rev2.id,
+        join: user in User,
+        on: post.user_id == user.id,
+        where:
+          post.section_id == ^section_id and post.user_id != ^user_id and
+            post.status in [:approved, :archived] and is_nil(rev2) and
+            rev.deleted == false,
```

Comment:
Same comment as above, this can be simplified to only join through SectionResource.